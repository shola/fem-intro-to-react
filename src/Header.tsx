import * as React from 'react';
import { Link } from 'react-router';
// const { Link } = ReactRouter;

interface SearchEventHandler {
    (e: any): void
}

interface HeaderProps {
    searchTerm?: string,
    handleSearchTermEvent?: SearchEventHandler,
    showSearch?: boolean
}
export default class Header extends React.Component<HeaderProps, {}> {
    render() {
        let utilSpace;
        if (this.props.showSearch) {
            utilSpace = (
                <input
                    type="text"
                    className="search-input"
                    placeholder="Search"
                    value={this.props.searchTerm}
                    onChange={this.props.handleSearchTermEvent}
                />
            );
        } else {
            utilSpace = (
                <h2 className="header-back">
                    <Link to="/search">
                        Back
                    </Link>
                </h2>
            );
        }
        return (
            <header className="header">
                <h1 className="brand">
                    <Link to="/" className="brand-link">
                        svideo
                    </Link>
                </h1>
                {utilSpace}
            </header>
        );
    }
}
