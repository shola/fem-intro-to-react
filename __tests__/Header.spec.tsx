import * as React from 'react';
import * as ReactDOM from 'react-dom';
import * as Renderer from 'react-test-renderer';
import Header from '../src/Header';

describe('Header View', () => {
    it('Check snapshot for unexpected changes', () => {
        const component = Renderer.create(<Header />);
        let tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });
});
