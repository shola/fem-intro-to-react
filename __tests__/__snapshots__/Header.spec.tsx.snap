// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header View Check snapshot for unexpected changes 1`] = `
<header
  className="header"
>
  <h1
    className="brand"
  >
    <a
      className="brand-link"
      onClick={[Function]}
      style={Object {}}
    >
      svideo
    </a>
  </h1>
  <h2
    className="header-back"
  >
    <a
      onClick={[Function]}
      style={Object {}}
    >
      Back
    </a>
  </h2>
</header>
`;
